### PR TITLE
[WIP] Parallel Maven release build

### PR DESCRIPTION
**Pull request by pierDipi**
Part of #596

## Proposed Changes

- Bump maven version for wrapper
- Parallel Maven build

### CODE REVIEW

**Comment by pierDipi** [comment]
Connection reset

/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

File: data-plane/docker/Dockerfile [hunk 1]
```diff
@@ -39,11 +39,11 @@ COPY docker/generate_jdk.sh .
 
 # Install dependencies. Note: don't build a single submodule (recever or dispatcher) since it just slows down
 # consecutive builds.
-RUN ./mvnw install -am -DskipTests -Drelease -Dlicense.skip -Deditorconfig.skip --no-transfer-progress
+RUN ./mvnw install -T 2C -am -Drelease -DskipTests -Dmaven.test.skip -ff -Dlicense.skip -Deditorconfig.skip --no-transfer-progress
 
 COPY . .
 
-RUN ./mvnw package -pl=${APP_DIR} -Drelease -am -DskipTests -Deditorconfig.skip --no-transfer-progress
+RUN ./mvnw package -pl=${APP_DIR} -T 2C -am -Drelease -DskipTests -Dmaven.test.skip -ff -Deditorconfig.skip --no-transfer-progress
```
Comment (slinkydeveloper) [inline data-plane/docker/Dockerfile:46]: No need to have both test skip variables. One of them is fine

Comment (slinkydeveloper) [inline data-plane/docker/Dockerfile:46]: Multithreaded builds, good catch!

Comment (pierDipi) [reply data-plane/docker/Dockerfile:46]: `-DskipTests` skips running tests whereas `-Dmaven.test.skip` skips compiling tests.

Comment (pierDipi) [reply data-plane/docker/Dockerfile:46]: Yeah, when it works, now it doesn't
```
[ERROR] Failed to execute goal on project receiver: Could not resolve dependencies for project dev.knative.eventing.kafka.broker:receiver:jar:1.0-SNAPSHOT: dev.knative.eventing.kafka.broker:core:jar:tests:1.0-SNAPSHOT was not found in https://oss.sonatype.org/content/repositories/snapshots/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of sonatype has elapsed or updates are forced -> [Help 1]
[ERROR] Failed to execute goal on project dispatcher: Could not resolve dependencies for project dev.knative.eventing.kafka.broker:dispatcher:jar:1.0-SNAPSHOT: dev.knative.eventing.kafka.broker:core:jar:tests:1.0-SNAPSHOT was not found in https://oss.sonatype.org/content/repositories/snapshots/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of sonatype has elapsed or updates are forced -> [Help 1]
```

Comment (slinkydeveloper) [reply data-plane/docker/Dockerfile:46]: If you say so, my understanding was that one is the alias of the other :smile:

Comment (slinkydeveloper) [reply data-plane/docker/Dockerfile:46]: should we close this pr then?

**Pull request closed without merging**